src/url-loader/url-resolver.ts

```diff
@@ -65,3 +70,25 @@ export abstract class UrlResolver {
     return url as ResolvedUrl;
   }
 }
+
+async function inferComponentDirname(dirname: string): Promise<string> {
+  try {
+    const contents = await fs.readFile(
+        pathlib.join(dirname, '.bowerrc'), {encoding: 'utf8'});
+    const bowerrc = JSON.parse(contents);
+    if (typeof bowerrc.directory === 'string') {
+      return bowerrc.directory;
+    }
+  } catch { /* don't care */
+  }
+  const [bowerJsonExists, packageJsonExists] = await Promise.all([
+    fs.exists(pathlib.join(dirname, 'bower.json')),
+    fs.exists(pathlib.join(dirname, 'package.json')),
+  ]);
+  if (bowerJsonExists) {
```

Comment:
I think we need to wire in handling for the case where we force npm or Bower. I'm about to add the --npm flag to polymer-build and friends, and in a future release we'll switch the default to npm. It'd be great if code at this level didn't have to change because the preference for the default and the force option were already passed down.